src/libraries/System.Private.CoreLib/src/System/StartupHookProvider.cs

```diff
@@ -27,13 +28,25 @@ private struct StartupHookNameOrPath
 
         // Parse a string specifying a list of assemblies and types
         // containing a startup hook, and call each hook in turn.
-        private static void ProcessStartupHooks()
+        private static void ProcessStartupHooks(string diagnosticStartupHooks)
         {
             if (!IsSupported)
                 return;
 
+            List<string> startupHookParts = new();
```

Comment:
This should avoid allocating the `List` if there are no startup hooks (the most common case).